### DISCUSSION TITLE
Best alerting/monitoring for Next

### DISCUSSION THREAD

**Discussion by ishaangandhi** [vercel/next.js#63772] [Help]
### Summary

Hi all!

I'm looking for the best way to monitor a Next app. I want to be able to:
* Get a push notification on my phone/desktop whenever there is a log of "error" on the server
* Be able to track client-side errors to a log
* See line-by-line execution times and overall execution time distributions for functions/routes.

I'm currently deployed on Vercel but I can change it if there is a better way of doing this.

What have people found to work well?

I have looked at sentry.io and DataDog so far.

**Comment by SouthLink** [comment]
This may require multiple tools to do what you're talking about:

1. When the mobile/desktop app receives a push notification when an error is detected, you may need [slack](https://slack.com/)
2. You can use Sentry to trace client errors to logs.
3. View the line-by-line and overall execution time distribution of functions/routes. You may need [sentry custom-instrumentation](https://docs.sentry.io/platforms/javascript/performance/instrumentation/custom-instrumentation/) 

When using slack, we usually use the [custom-error](https://nextjs.org/docs/pages/building-your-application/routing/custom-error) to monitor for errors that can cause whitescreens or crash levels in the application, and report the errors to sentry with a special tag, otherwise countless errors will crash you。

**Reply by aziyatali** [reply]
Agreed, configure Sentry with Nextjs and Slack.
